packages/design-system/src/styles/components/_Button.scss

```diff
@@ -90,207 +79,136 @@
-
-  &:hover,
-  &.ds-c-button--hover {
-    border-color: rgba($border-color-inverse, 0.8);
-    color: rgba($color-base-inverse, 0.8);
-  }
-
-  &:focus,
-  &.ds-c-button--focus {
-    @if $ds-include-focus-styles {
-      @include focus-styles;
-    } @else {
-      border-color: rgba($border-color-inverse, 0.8);
-      color: rgba($color-base-inverse, 0.8);
-    }
-  }
-
-  &:active,
-  &.ds-c-button--active {
-    border-color: rgba($border-color-inverse, 0.6);
-    color: rgba($color-base-inverse, 0.6);
-  }
-  @if $ds-include-focus-styles {
-    // Make sure active overrides focus styles
-    &:active:focus,
-    &.ds-c-button--active:focus {
-      background-color: $color-background-inverse;
-      color: rgba($color-base-inverse, 0.6);
-    }
-  }
-}
-
-/* stylelint-disable scss/at-extend-no-missing-placeholder */
-%transparent-inverse-button {
-  background-color: transparent;
-  border-color: transparent;
-
-  // Default inverse focus styles are overridden above
-  @if $ds-include-focus-styles {
-    &:focus,
-    &.ds-c-button--focus {
-      @include focus-styles;
-    }
-  }
-}
-
-.ds-c-button--inverse.ds-c-button--transparent,
-.ds-c-button--inverse.ds-c-button--transparent:visited {
-  @extend %transparent-inverse-button;
-  &:disabled {
-    background-color: $color-gray-lighter;
-    border-color: transparent;
-    color: $color-gray-dark;
-  }
-}
-
-/**
- *  Not recommended usage
- *  Equivalent to ".ds-c-button--inverse.ds-c-button--transparent"
- */
-.ds-c-button--transparent-inverse,
-.ds-c-button--transparent-inverse:visited {
-  @extend .ds-c-button--inverse;
-  @extend %transparent-inverse-button;
-}
+ *  Status variants
 
-%disabled-inverse-button {
-  background-color: $color-background-inverse;
-  border-color: $color-gray-lighter;
-  color: $color-gray-lighter;
-  pointer-events: none;
-}
+Default
+1. button bg
+2. button bg-hover
+3. button bg-active
+4. button text color
+5. button border color
+6. button border color-hover
+7. button border color-active
+8. button text color-hover
+9. button text color-active
+10. button bg-disabled
+11. buton text color-disabled
+12. button border color-disabled
+13. button text decoration
+*/
 
-.ds-c-button--inverse:disabled,
-.ds-c-button--inverse.ds-c-button--disabled {
-  @extend %disabled-inverse-button;
-}
-
-/**
- *  Not recommended usage
- *  Equivalent to ".ds-c-button--inverse.ds-c-button--disabled"
- */
-.ds-c-button--disabled-inverse,
-.ds-c-button--disabled-inverse:visited {
-  @extend .ds-c-button--inverse;
-  @extend %disabled-inverse-button;
-}
-/* stylelint-enable scss/at-extend-no-missing-placeholder */
-
-/**
- *  Status variants
- */
 $button-color-variants: (
-  primary: $button-primary-bg $button-primary-bg--hover $button-primary-bg--active,
-  success: $color-success $color-success-dark $color-success-darker,
-  danger: $color-error $color-error-dark $color-error-darkest,
+  button: $button-default-bg $button-default-bg--hover $button-default-bg--active
```

Comment:
[question] Is it standard for variables like this to use lists like this and then reference the list position in the styles below?

Do we ever do something more explicit like
```
$button-color-variants: (
  button: (
    background-color: $button-default-bg,
    background-hover: $button-default-bg--hover,
    background-active: $button-default-bg--active
    ...
  )
)
```
and then reference it below like `map-get($color, "background-color")`?

I'm rusty on sass so I don't know if:
1. that's possible
2. that's a convention we use in other places in the project and is something to do here
3. that's something worth doing to be more explicit instead of relying on list position 🤷🏻‍♀️

Reply:
I'd definitely be interested in simplifying and making this more readable - I was extending what was used previously. I can look to see if there is a way we can make it more readable

Reply:
Okay! And if it's not possible or it's taking hours to refactor, it's not a big deal. 🙂